lib/components/Checkbox/Checkbox.tsx

```diff
@@ -5,12 +5,42 @@ import {
   InlineFieldProps,
 } from '../private/InlineField/InlineField';
 
+const resolveCheckedGroup = (values: Array<CheckboxChecked>) =>
+  values.reduce((acc, val, index) => {
+    if (acc === 'mixed') {
+      return 'mixed';
+    }
+
+    if (acc === true) {
+      return val || 'mixed';
+    }
+
+    if (acc === false && index > 0) {
+      return val ? 'mixed' : false;
+    }
+
+    return val;
+  }, false);
```

Comment:
Sorry to jump in, but it did make me think that this could be shortened to something like:

```suggestion
  values.some(val => val !== values[0]) ? 'mixed' : a[0] ?? false;
```

Reply:
Sometimes you just stare at the same code for too long 😄 .

Thanks mate, i like the refactor 👏